b2sdk/transfer/inbound/downloader/parallel.py

```diff
@@ -49,14 +51,23 @@ class ParallelDownloader(AbstractDownloader):
     #
     FINISH_HASHING_BUFFER_SIZE = 1024**2
 
-    def __init__(self, max_streams, min_part_size, *args, **kwargs):
+    def __init__(
+            self,
+            max_streams: int,
+            min_part_size: int,
+            max_download_workers: int = 10,
+            *args,
+            **kwargs,
+    ) -> None:
         """
         :param max_streams: maximum number of simultaneous streams
         :param min_part_size: minimum amount of data a single stream will retrieve, in bytes
+        :param int max_download_workers: maximum number of download threads
         """
+        super().__init__(*args, **kwargs)
+        LazyThreadPool.__init__(self, max_workers=max_download_workers)
```

Comment:
I think this is weird, super() calls should chain each other, should not need you to call two explicitly

Reply:
I believe with multiple inheritance, `super()` only calls the first parent class' constructor.

Reply:
yes, but that parent should call it parent, and that one should call its own parent etc, all the way to `object`

Reply:
If suspect you're missing the point there's multiple inheritance here. `LazyThreadPoolMixin` is **not** a parent of  `AbstractDownloader` so where is its `__init__` supposed to be called?

Reply:
please watch https://www.youtube.com/watch?v=EiOglTERPEo (billing `RT / onboarding`) and then lets talk again

Reply:
:tv: :eyes:

Reply:
Thanks for sharing this. I've learned quite a bit. I'll change this to avoid calling `LazyThreadPool.__init__` explicitly.